### DISCUSSION TITLE
Dynamic foreign key

### DISCUSSION THREAD

**Discussion by magrinj** [prisma/prisma#11676] [Q&A] [answered]
Hey folks,

I'm actually working on a projet where I've abstracted some models like Movie and TV-Show, but each of this model can be link to Gender, Actor, ProductionCompany, Post etc...
I don't want to store a `movieId` and a `tvShowId` key in the models for each subtype, and I just want to have one foreign key for both model.
So for that I'm creating an `Entity` model as an intermediate holding and `entityId` key that can be a Movie or a TV-Show id, and an `entityType` enum saying if it's a Movie or a TV-Show foreign key.
From what I'v read about Prisma schema I don't really know if it's possible to create relation of this kind, as `@relation` only link one model to another.
Does anyone already faced this issue and found a workaround to do that ?

Thanks ! :)

**Comment by nurul3101** [comment, marked as answer]
Hey @magrinj 👋 

Currently it's not possible to have dynamic foreign keys, while defining a relation attribute prisma needs to know the referenced model.
As of now in this case `entityId` column having either `movieId` or `showId` won't be feasible, two separate columns named `moveId` and `showId` would need to be created.
We have a related Feature Request for [Supporting Union Type](https://github.com/prisma/prisma/issues/2505) which might be helpful in this case. If it looks something that could help you please feel free to give a 👍 to the issue so that our engineering and product team could prioritise it.

**Comment by Apoorve73** [comment]
@nurul3101 I have come across a similar case as @magrinj. Can you please tell me if this feature of Dynamic Association now available in Prisma?

**Reply by nurul3101** [reply]
Hey,

No, it's not possible at the moment. You would still need to separate out the two columns, dynamic foreign key isn't supported yet.

**Reply by Apoorve73** [reply]
Thanks for the prompt response @nurul3101! Is it being worked upon or planned for this year's roadmap?

**Reply by nurul3101** [reply]
It's not on the immediate roadmap yet. You can check out our roadmap here: [Prisma Roadmap](https://prismaio.notion.site/Prisma-Roadmap-50766227b779464ab98899accb98295f).

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
